http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -669,9 +683,21 @@ class LeonardoService(
                                   throw403 = true)
 
       runtimeConfig <- RuntimeConfigQueries.getRuntimeConfig(cluster.runtimeConfigId).transaction
-      now <- timer.clock.realTime(TimeUnit.MILLISECONDS).map(Instant.ofEpochMilli)
-      _ <- CloudService.Dataproc.interpreter
-        .stopRuntime(StopRuntimeParams(RuntimeAndRuntimeConfig(cluster, runtimeConfig), now))
+      _ <- if (runtimeConfig.cloudService == CloudService.Dataproc) IO.unit
+      else IO.raiseError(CloudServiceNotSupportedException(runtimeConfig.cloudService))
+
+      // throw 409 if the cluster is not stoppable
+      _ <- if (cluster.status.isStoppable) IO.unit
+      else
+        IO.raiseError[Unit](RuntimeCannotBeStoppedException(cluster.googleProject, cluster.runtimeName, cluster.status))
+
+      // Update the cluster status to Stopping in the DB
+      now <- nowInstant
+      _ <- clusterQuery.setToStopping(cluster.id, now).transaction
+
+      // stop the runtime
+      traceId <- ev.ask
+      _ <- publisherQueue.enqueue1(StopRuntimeMessage(cluster.id, Some(traceId)))
```

Comment:
will this impact ui in anyway? since the api will be changed from sync to async?

Reply:
I don't think so.. Stopping was always async, now we just do the Google calls in back leo instead of front Leo. The API and DB behavior should be the same.

I am testing with Terra UI anyway though..